lighthouse-core/audits/preload-lcp.js

```diff
@@ -0,0 +1,185 @@
+   * @param {LH.Artifacts.TraceElement|undefined} lcpElement
+   * @return {string|null}
+   */
+  static getURLToPreload(mainResource, graph, lcpElement) {
+    if (!lcpElement) {
+      return null;
+    }
+
+    let lcpUrl = null;
+    graph.traverse((node, traversalPath) => {
+      if (node.type !== 'network') return;
+      const record = node.record;
+      const imageSource = lcpElement.imageSource;
+      // Don't include the node itself or any CPU nodes in the initiatorPath
+      const path = traversalPath.slice(1).filter(initiator => initiator.type === 'network');
+      if (!PreloadLCPAudit.shouldPreloadRequest(record, mainResource, path, imageSource)) return;
+      lcpUrl = node.record.url;
+    });
+
+    return lcpUrl;
+  }
+
+  /**
+   * Computes the estimated effect of preloading the LCP image.
+   * @param {string} lcpUrl // The image URL of the LCP
+   * @param {LH.Gatherer.Simulation.GraphNode} graph
+   * @param {LH.Gatherer.Simulation.Simulator} simulator
+   * @return {{wastedMs: number, results: Array<{url: string, wastedMs: number}>}}
+   */
+  static computeWasteWithGraph(lcpUrl, graph, simulator) {
+    const simulation = simulator.simulate(graph, {flexibleOrdering: true});
+
+    /** @type {LH.Gatherer.Simulation.GraphNode|null} */
+    let lcpNode = null;
+    /** @type {LH.Gatherer.Simulation.GraphNode|null} */
+    let mainDocumentNode = null;
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+
+      const networkNode = /** @type {LH.Gatherer.Simulation.GraphNetworkNode} */ (node);
+      if (node.isMainDocument()) {
+        mainDocumentNode = networkNode;
+      } else if (networkNode.record && lcpUrl === networkNode.record.url) {
+        lcpNode = networkNode;
+      }
+    });
+
+    if (!mainDocumentNode) {
+      // Should always find the main document node
+      throw new Error('Could not find main document node');
+    }
+
+    if (!lcpNode) {
+      // Should always find the LCP node if we've gotten to this point
+      throw new Error('Could not find the LCP node');
+    }
+
+    const timing = simulation.nodeTimings.get(lcpNode);
+    if (!timing) throw new Error('Missing LCP node');
+    return {
+      wastedMs: timing.duration,
+      results: [
+        {wastedMs: timing.duration, url: lcpUrl},
+      ],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const trace = artifacts.traces[PreloadLCPAudit.DEFAULT_PASS];
+    const devtoolsLog = artifacts.devtoolsLogs[PreloadLCPAudit.DEFAULT_PASS];
+    const URL = artifacts.URL;
+    const simulatorOptions = {trace, devtoolsLog, settings: context.settings};
+    const lcpElement = artifacts.TraceElements
+      .find(element => element.traceEventType === 'largest-contentful-paint');
+
+    /** @type {LH.Config.Settings} */
+    // @ts-expect-error
```

Comment:
we should use the real settings from `context.settings` (just reuse `simulatorOptions`)